libs/domain/product/src/services/adapter/normalizers/facet-rating/facet-rating.normalizer.ts

```diff
@@ -0,0 +1,60 @@
+import { Transformer } from '@spryker-oryx/core';
+import { resolve } from '@spryker-oryx/di';
+import { RouterService } from '@spryker-oryx/router';
+import { tap } from 'rxjs';
+import { ApiProductListModel, Facet } from '../../../../models';
+
+export const FacetRatingNormalizer = 'oryx.FacetRatingNormalizer*';
+
+export function facetRatingNormalizer(
+  ratingFacet: ApiProductListModel.RangeFacet
+): Facet {
+  const ratingParamKey = 'rating[min]';
+
+  const routerService = resolve(RouterService);
+
+  let selectedValues: string[] = [];
+  routerService
+    .currentQuery()
+    .pipe(
+      tap((params) => {
+        selectedValues = params?.[ratingParamKey]
+          ? [params?.[ratingParamKey] as string]
+          : [];
+      })
+    )
+    .subscribe()
```

Comment:
We can use observables as result so not needed doint subscribe>unsubsribe and rely on result of subscription for other functionality